src/hooks/useTokenInfo.ts

```diff
@@ -0,0 +1,145 @@
+import { useEffect, useState } from 'react'
+import { getERC20TokenContract } from '~/getters'
+import { getPublicWeb3Provider } from '~/shared/stores/wallet'
+
+interface TokenInfo {
+    symbol: string
+    decimals: number
+    name: string
+}
+
+function entryKey(tokenAddress: string, chainId: number) {
+    return `${chainId}-${tokenAddress}`
+}
+
+const tokenInfoCache: Partial<
```

Comment:
how about:
```
const tokenInfoCache: Map<string, TokenInfo | null | Promise<TokenInfo | null>> = new Map()
```
?
Map does pretty much the same and is a bit more readable than Partial<Record<SomethingElse<LetsDiveDeeper<...>>>>

Reply:
I'd really have to want to use a map. I don't see much of a benefit, really, do you?

- Keys are predefined and are always strings.
- Order does not matter.

I'll make the current notation better though!

```ts
const tokenInfoCache: Record<
    string,
    TokenInfo | null | Promise<TokenInfo | null> | undefined
> = {}
```

Pretty much the same in our case and reads much better.

Reply:
It's not that I strongly advocate for using Map - for simple objects it's just a bit better in my opinion cause the code looks a bit cleaner when you use stuff like `map.has('someProperty')` or `map.get('someProperty')` or `map.delete('someProperty')`. Anyways - now it looks a bit cleaner so we're good :)

Reply:
Good input!